src/msix/AppxPackageObject.cpp

```diff
@@ -245,16 +246,21 @@ namespace MSIX {
 
             for (const auto& package : bundleInfo->GetPackages())
             {
-                auto bundleInfoInternal = package.As<IAppxBundleManifestPackageInfoInternal>();
-                auto packageName = bundleInfoInternal->GetFileName();
-                auto packageStream = m_container->GetFile(Encoding::EncodeFileName(packageName));
+				LPWSTR packageName = nullptr;
+				auto bundleInfoInternal = package.As<IAppxBundleManifestPackageInfo>();

```

Comment:
>IAppxBundleManifestPackageInfo [](start = 41, length = 30)

Why did you change to this interface?  Seems like it just made things harder here.

Reply:
My suggestion would be to leave this all alone and simply .As it right before you need it below.

---
In reply to: [288749576](https://github.com/Microsoft/msix-packaging/pull/146#discussion_r288749576) [](ancestors = 288749576)